### PR TITLE
543 Refactor of Saved Search Accordion

### LINKED ISSUE #543: Refactor of Saved Search Accordion
# 💡 Summary #

- The creation of the new Side Drawer with its use of Text Fields and toggle-able Checkboxes makes the current Data Grid based component feel out of place.
- Plus there needs to be a way to quickly toggle a Saved Search off and on, prevent duplicate names and very general searches.
- Given this and the nature of the data displayed, a conversion of the table into a Checkbox/Radio controlled list would be a good step forward.

## Motivation and context ##
- Resolve the aforementioned bugs. 
- Consistent User Experience.
- Reduced footprint.

## Implementation notes ##

Screenshot 2024-08-15 at 10 38 25 AM

Screenshot 2024-08-15 at 10 52 07 AM

## Acceptance criteria ##

How do we know when this work is done?

- [x] New Checkbox controlled list is implemented.
- [x] It displays a list of user created Saved Searches.
- [ ] Checked entries indicates that the selected Search is being used.
- [ ] Unchecked entries indicates the item is not in use.
- [ ] An icon to delete each Saved Search.

### PR DESCRIPTION
**Pull request by hawkishpolicy**
#  #
Refactored the Saved Searches Accordion to better match the recently improved Side Drawer.
## 🗣 Description ##
- Removed the old MUI Data Grid based table.
- Saved Searches are now displayed via a series of Radio button controlled list items.
- The user can now toggle the active status of the Saved Search.
- Updated the Save Search button to a MUI button.
- Provided a name check to prevent the User from saving duplicate named searches.
- Provided a filter check to prevent User from saving a search with no advanced filters.
- Removed fetchSearch call, savedSearch state, and savedSearchCount state from Dashboard.
- These items are now in their own context.
- Extended context to Drawer Interior and Dashboard.

## 💭 Motivation and context ##
- Closes #543 
- Increases Saved Search functionality and brings its design in with the rest of the Side Drawer.

## 📷 Screenshots (if appropriate) ##
Screenshot 2024-08-15 at 10 38 25 AM
Screenshot 2024-08-23 at 5 14 15 PM
Screenshot 2024-08-23 at 4 55 29 PM
Screenshot 2024-08-23 at 5 15 16 PM

Screenshot 2024-08-23 at 5 14 34 PM

Screenshot 2024-08-23 at 4 55 29 PM
Screenshot 2024-08-23 at 4 55 35 PM
## ✅ Pre-approval checklist ##

- [x] This PR has an informative and human-readable title.
- [x] Changes are limited to a single goal - *eschew scope creep!*
- [x] *All* future TODOs are captured in issues, which are referenced
      in code comments.
- [x] All relevant type-of-change labels have been added.
- [x] I have read the [CONTRIBUTING](../blob/develop/CONTRIBUTING.md) document.
- [x] These code changes follow [cisagov code standards](https://github.com/cisagov/development-guide).
- [x] All new and existing tests pass.

## ✅ Pre-merge checklist ##

- [ ] Revert dependencies to default branches.
- [ ] Finalize version.

## ✅ Post-merge checklist ##

- [ ] Create a release.

### CODE REVIEW

File: frontend/src/components/DrawerInterior.tsx [hunk 1]
```diff
@@ -23,12 +26,12 @@ import {
   FiberManualRecordRounded
 } from '@mui/icons-material';
 import FilterAltIcon from '@mui/icons-material/FilterAlt';
+import SaveIcon from '@mui/icons-material/Save';
```
Comment (ameliav) [inline frontend/src/components/DrawerInterior.tsx]: Combine the import of the newly added icons.
```suggestion
  FiberManualRecordRounded,
  FilterAlt as FilterAltIcon,
  Save as SaveIcon
} from '@mui/icons-material';
```

File: frontend/src/components/OrganizationSearch.tsx [hunk 2]
```diff
@@ -65,6 +64,7 @@ export const OrganizationSearch: React.FC<OrganizationSearchProps> = ({
   // const [tags, setTags] = useState<OrganizationTag[]>([]);
```
Comment (ameliav) [inline frontend/src/components/OrganizationSearch.tsx]: Can go ahead and remove unused comments and code. (ie. lines 3 - 4, 63 - 64, 135 - 137)
```suggestion
```

**Review by ameliav — CHANGES_REQUESTED** [review]
Found a few code maintenance items, nothing big though

File: frontend/src/components/DrawerInterior.tsx (hunk 1 above)
Comment (chrtorres) [reply frontend/src/components/DrawerInterior.tsx]: This has been fixed.

File: frontend/src/components/OrganizationSearch.tsx (hunk 2 above)
Comment (chrtorres) [reply frontend/src/components/OrganizationSearch.tsx]: Unused comments have been removed.

**Review by ameliav — APPROVED** [review]
LGTM

**Review by chrtorres — APPROVED** [review]

**Review by schmelz21 — APPROVED** [review]

**Pull request merged**